src/artifacts_poetry_plugin/plugin.py

```diff
@@ -9,143 +9,234 @@
+        username = None
+        password = None
+        if auth:
+            username = auth.username
+            password = auth.password
         else:
-            non_deployable_packages.append(package)
-    return deployable_packages, non_deployable_packages
-
-
-def upload_packages(poetry, packages: List[Package], url, username, password, io):
-    uploader = Uploader(poetry=poetry, io=io)
-    uploader.auth(username=username, password=password)
-    session = uploader.make_session()
-    for package in packages:
-        uploader._package = package
-        for file in package.files:
-            if "url" in file:
-                uploader._upload_file(
-                    session,
-                    url,
-                    file=file["url"],
-                )
-
-
-def add_project_package(poetry, packages: List[Package]):
-    """Adds the generated package to the list of packages with its associated wheel file"""
-    project_package = poetry.package
-    for root, dirs, files in os.walk("dist/"):
-        for file in files:
-            if file.endswith(".whl"):
-                file_data = {"file": file, "url": Path(os.path.join(root, file))}
-                project_package.files.append(file_data)
-    packages.append(project_package)
-
-
-def get_deploy_packages(poetry, cache_dir):
-    """Gets all deployable packages from the given cache directory. A package is deployable if it has any wheel files associated with it"""
-    packages = get_packages(poetry)
-    wheel_files = get_cached_wheel_files(cache_dir)
-    deployable_packages, non_deployable_packages = add_package_paths(
-        packages=packages, wheel_files=wheel_files
-    )
-    add_project_package(poetry, deployable_packages)
-    return deployable_packages, non_deployable_packages
-
-
-class ArtifactsDeploy(Command):
+            raise RuntimeError(err_msg)
+        if username is None or password is None:
+            raise RuntimeError(err_msg)
+        return RepoData(url, username, password)
+
+    def get_poetry_packages(self, poetry) -> List[Package]:
+        """Gets packages from the given poetry.lock file"""
+        locker = poetry.locker
+        repository = locker.locked_repository()
+        return repository.packages
+
+    def get_cached_wheel_files(self, cache_dir):
+        """Gets all wheel file locations for the given cache directory"""
+        wheel_files = {}
+        for root, dirs, files in os.walk(cache_dir):
+            for file in files:
+                if file.endswith(".whl") or file.endswith(".tar.gz"):
+                    wheel_files[file] = Path(os.path.join(root, file))
+        return wheel_files
+
+    def get_deploy_non_deploy_packages(self, packages: List[Package], wheel_files):
+        """Adds the wheel file locations for each package's files"""
+        non_deployable_packages = []
+        deployable_packages = []
+        for package in packages:
+            file_found = False
+            for file in package.files:
```

Comment:
assuming file is some kind of map that is returned by package.files, I'd name it something a bit more descriptive

Reply:
renamed to file_info